swarm/src/handler/either.rs

```diff
@@ -64,6 +64,29 @@ where
     }
 }
 
+// Taken from https://github.com/bluss/either.
+impl<L, R> IntoEitherHandler<L, R> {
```

Comment:
Do we still need `IntoEitherHandler`?

Reply:
How else would the `IntoConnectionHandler::into_handler` methods of the left and right branch be called?

Reply:
What I was getting at is: Can we replace it with `Either`?

Reply:
I think we discussed this in the past, see https://github.com/libp2p/rust-libp2p/pull/2192#issuecomment-900213835 @thomaseizinger.

Reply:
You've got much better memory than I do :D

Reply:
What do you think about adding a comment about it?